### PR TITLE
Fix build on illumos

### PR DESCRIPTION
**Pull request by pyhalov**
DESCRIPTION: Fix build on illumos

### CODE REVIEW

**Comment by pyhalov** [comment]
alloca.h is necessary to get alloca() on illumos/Solaris, sys/stat.h is necessary to get S_IRUSR, S_IWUSR definitions.

File: src/backend/distributed/shared_library_init.c [hunk 1]
```diff
@@ -12,6 +12,9 @@
 #include <limits.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#ifdef __sun
+#include <alloca.h>
+#endif
```
Comment (marcocitus) [inline src/backend/distributed/shared_library_init.c]: Could you add your comment here? 

> alloca.h is necessary to get alloca() on illumos/Solaris, sys/stat.h is necessary to get S_IRUSR, S_IWUSR definitions.

Comment (pyhalov) [reply src/backend/distributed/shared_library_init.c]: Updated PR

**Comment by marcocitus** [comment]
Merged into master.

**Pull request closed without merging**